source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,166 @@
+=================
+Connection Storms
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. _connection-tuning-java:
+
+Overview
+--------
+
+In this guide, you can learn about **connection storms** and settings that
+can help avoid them.
+
+Connection Storms
+-----------------
+
+A **connection storm** is an event in which application servers
+start an increasing number of connection requests, but the host server
+CPU is unable to schedule them.
+
+.. example::
+
+   Consider an application that uses MongoDB, and is served from an
+   auto scaling service. A sudden spike in traffic could cause the scaling
+   service to stand up many new application instances. This sudden burst
+   in application instances attempting to establish connections to MongoDB
+   is one example of a connection storm.
+
+Common Causes and Effects
++++++++++++++++++++++++++
+
+Common causes for connection storms in MongoDB include:
+
+- Incorrect application behavior
+- Network latency
+- Insufficient hardware for the workload
+- Failover events
+
+In MongoDB, connection storms can have multiple effects. For example:
+
+- The high volume of connection requests leads to latency in processing.
+- Replication connections between nodes may be blocked or slowed, so replication
+  operations cannot proceed.
+- Administrative connections to the database are blocked, preventing
+  a user from logging in.
+- Monitoring connections to the database are blocked, degrading visibility
+  into the problem.
+- CPU or memory oversubscription due to connection management can lead
+  to a ``mongod`` crashing and restarting. This extra workload on the server
+  can be exclusively attributed to connection handling.
+- In Atlas, connection requests can exceed instance tier limits, causing
+  refusal of new connections.
```

Comment:
I suggest an alternate framing here, which peels out the activity (Rate of connections vs. Total number of connections) and then describes the possible effects of either of them together (which are similar), with less technical detail about how the effects can unfold. Should help future-proof these docs as improvements are made over time.

I don't have the right link formatting 😊 

```suggestion
The effects of connection storms follow from either:

- the performance impact of the rate at which new connections are being made. Rapidly incoming connection requests can cause bottlenecks during connection establishment. Since it is common to retry failed connections, connection storms can persist until some load is shed.
- inability to connect to a database node when that node's [maxIncomingConnections|https://docs.mongodb.com/manual/reference/configuration-options/#mongodb-setting-net.maxIncomingConnections] setting is reached.

Either of these concerns can lead to practical issues like:

- Hardware overuse, including heavy queuing, crashing and restarting due to lack of memory, etc.
- Replica set failover due to interference with intra-node system operations.
- Inability to connect to and diagnose an issue on the database side.
- Loss of connectivity for monitoring tools and other support services.
- Interference with monitoring and administration activities.
```